src/mainwindow.cpp

```diff
@@ -234,6 +236,23 @@ void MainWindow::setup_ActionsMenu_Actions()
     setup_Action(MOVE_RIGHT, new QAction(tr("Move Tab &Right"), settingOwner),
                  MOVE_RIGHT_SHORTCUT, consoleTabulator, SLOT(moveRight()), menu_Actions, data);
 
+    menu_Actions->addSeparator();
+    data.setValue(checkHasIndexedTab);
+
+    const QString textBase = tr("Tab");
+    QSignalMapper *signalMapper = new QSignalMapper(this);
+    QObject::connect(signalMapper, SIGNAL(mapped(int)), consoleTabulator, SLOT(switchTab(int)));
+    QObject::connect(signalMapper, SIGNAL(mapped(int)), this, SLOT(gotshit(int)));
+    for (int i=1; i<=10; ++i) {
+        QString num = QString::number(i);
+        QAction *action = new QAction(textBase + QLatin1Char(' ') + num, settingOwner);
+        action->setProperty("tab", i);
+        char name[16];
+        snprintf(name, sizeof(name), "Tab %d", i);
+        signalMapper->setMapping(action, i - 1);
+        setup_Action(name, action, NULL, signalMapper, SLOT(map()), menu_Actions, data);
```

Comment:
Could you move `Tab X` actions to a submenu (e.g., a submenu `Go to`). After this patch the actions menu appears bloated.

Reply:
Done.